examples/async.rs

```diff
@@ -1,16 +1,25 @@
 extern crate futures;
 extern crate reqwest;
 extern crate tokio_core;
+extern crate tokio_io;
 
+use std::io;
 use futures::Future;
 
 fn main() {
     let mut core = tokio_core::reactor::Core::new().unwrap();
     let client = reqwest::unstable::async::Client::new(&core.handle()).unwrap();
 
-    let work = client.get("https://hyper.rs").unwrap().send().map(|res| {
-        println!("{}", res.status());
-    });
+    let work = client.get("http://localhost:8080/range/4096?duration=5&chunk_size=300")
```

Comment:
I could see adding something like `chunk_size: 300` as part of the `server!` macro, to change from using `write_all` on the whole response to looping and trying to write that number of bytes at a time.

Usage of the macro could look like this:

```rust
let server = server! {
    request: expected_req,
    response: bytes_to_respond_with,
    chunk_size: 300
};
```

What does `duration` do? Is that a timeout between chunks?

Implementation in the `support/server.rs` would need these things:

- add `chunk_size: Option<usize>` to the `Txn` struct
- Update the [write portion[(https://github.com/seanmonstar/reqwest/blob/master/tests/support/server.rs#L57-L64) to loop on writing `if let Some(size) = self.chunk_size`.